atc/creds/ssm/ssm.go

```diff
@@ -99,5 +112,63 @@ func (s *Ssm) getParameterByPath(path string) (interface{}, *time.Time, bool, er
 	if len(value) == 0 {
 		return nil, nil, false, nil
 	}
+
 	return value, nil, true, nil
 }
+
+func isValidFormat(format string) bool {
+	switch format {
+	case "raw", "trim", "yml", "yaml", "json":
+		return true
+	}
+	return false
+}
+
+func (s *Ssm) getParameterFormat(id *string) (string, error) {
+	format := "raw"
+
+	tags, err := s.api.ListTagsForResource(&ssm.ListTagsForResourceInput{
+		ResourceType: aws.String(ssm.ResourceTypeForTaggingParameter),
+		ResourceId:   id,
+	})
+
+	if err == nil {
+		for i := range tags.TagList {
+			tag := strings.ToLower(*tags.TagList[i].Key)
+			if tag == "format" {
+				format = strings.ToLower(*tags.TagList[i].Value)
+			}
+		}
+
+		if !isValidFormat(format) {
+			return "nil", fmt.Errorf("invalid format %s", format)
+		}
+	}
+
+	return format, nil
+}
+
+func unmarshalParameterFormat(input string, format string) (interface{}, error) {
+	var value interface{}
+
+	switch format {
+	case "json":
+		err := json.Unmarshal([]byte(input), &value)
+		if err != nil {
+			return nil, err
+		}
+	case "yml", "yaml":
+		err := yaml.Unmarshal([]byte(input), &value)
+		if err != nil {
+			return nil, err
+		}
+	case "trim":
+		value = strings.TrimSpace(string(input))
+	case "raw":
+		value = string(input)
```

Comment:
```suggestion
		value = input
```
Redundant type casting